src/transformers/generation/utils.py

```diff
@@ -667,6 +667,9 @@ def _prepare_decoder_input_ids_for_generation(
         # no user input -> use decoder_start_token_id as decoder_input_ids
         if decoder_input_ids is None:
             decoder_input_ids = decoder_input_ids_start
+        # exception: Donut checkpoints have task-specific decoder starts and don't expect a BOS token
+        elif self.config.model_type == "vision-encoder-decoder" and "donut" in self.name_or_path.lower():
+            pass
```

Comment:
Found this approach more readable than adding an `elif not (condition) and ...` below :)